### PR TITLE
Use Calico's own fork of kubernetes client-go instead of mine.

### PR DESCRIPTION
**Pull request by fasaxc**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Review by caseydavenport — APPROVED** [review]
LGTM, but looks like generated files are out of date?

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
